website/project/model.py

```diff
@@ -1411,6 +1411,35 @@ def set_node_license(self, license_id, year, copyright_holders, auth, save=False
         if save:
             self.save()
 
+    def subscribe_user_to_notifications(self, user):
+        """ Update the notification settings for the creator or contributors
+
+        :param user: User to subscribe to notifications
+        """
+        from website.notifications.utils import to_subscription_key
```

Comment:
One additional benefit of this approach is that it would allow us to get rid of the `ENABLE_NOTIFICATION_SUBSCRIPTION_CREATION` setting, which seems to exist solely for testing.

Our base test class already has a way to disconnect signals, so you'd be able to do something like:

```python
class NotificationTestCase(OsfTestCase):
    DISCONNECTED_SIGNALS = {
        contributor_added: [notify_added_contributor, subscribe_contributor]
        project_created: [subscribe_creator]
    }
```